app/controllers/dashboards_controller.rb

```diff
@@ -18,12 +18,13 @@ def show
 
   private
 
-  def authenticate_with_api_key
-    api_key   = request.headers["Authorization"] || params.permit(:api_key).fetch(:api_key, "")
-    @api_user = User.find_by_api_key(api_key)
+  def find_api_key
+    params_key = request.headers["Authorization"] || params.permit(:api_key).fetch(:api_key, "")
```

Comment:
This functionality encourages users to expose their API key to news reader services and apps like Feedbin, Feedly etc.

How can this be made safe so the secret used to protect this feed is only ever good for this and can't ever be used for more sensitive operations? Perhaps use a dedicated secret for this feed that is stored separately from API keys entirely? Or if an API token has the `show_dashboard` scope, don't allow it to have any other scopes?

Reply:
Another alternative might be to only accept the key as a header, and ignore the query string params or respond with an error if an api key query string param is detected. I don't expect news reader services to allow users to send custom headers.

Reply:
> only accept the key as a header

It would be preferred to release this without making any breaking change to existing functionality, even if they were insecure, to begin with. Any breaking change should start with an issue and only if there is agreement on its acceptability we can go ahead with it.
> if an API token has the show_dashboard scope, don't allow it to have any other scopes?

this is doable. All new keys will show validation error if `show_dashboard` is enabled with any other scope.